src/Parser/GLTFParser.js

```diff
@@ -0,0 +1,158 @@
+                delete obj.gltfShader.boundUniforms[name];
+            }
+        }
+    }
+}
+
+/**
+ * @module GLTFParser
+ */
+/**
+ * Enable Draco decoding for gltf.
+ * @param {string} path to draco library folder.
+ * This library is mandatory to load b3dm with Draco compression.
+ * @param {object} config optional configuration for Draco compression.
+ *
+ * The Draco library files are in folder itowns/examples/libs/draco/.
+ * You are obliged to indicate this path when you want enable the Draco Decoding.
+ *
+ * For more information on Draco, read file in /itowns/examples/libs/draco/README.md.
+ *
+ * @example <caption>Enable draco decoder</caption>
+ * // if you copied the folder from /itowns/examples/libs/draco/ to your root project,
+ * // You could set path with './'.
+ * itowns.enableDracoLoader('./');
+ */
+export function enableDracoLoader(path, config) {
+    if (!path) {
+        throw new Error('Path to draco folder is mandatory');
+    }
+    const dracoLoader = new DRACOLoader();
+    dracoLoader.setDecoderPath(path);
+    if (config) {
+        dracoLoader.setDecoderConfig(config);
+    }
+    glTFLoader.setDRACOLoader(dracoLoader);
+}
+
+export default {
+    /** Parse gltf buffer and extract THREE.Scene and batch table
+     * @param {ArrayBuffer} buffer - the gltf buffer.
+     * @param {Object} options - additional properties.
+     * @param {Matrix4=} [options.gltfUpAxisMatrix={}] - Matrix4f up axis transformation, by default Z->Y is applied
+     * @param {string} options.urlBase - the base url of the glTF file (used to fetch textures for the embedded glTF model).
+     * @param {boolean=} [options.doNotPatchMaterial='false'] - disable patching material with logarithmic depth buffer support.
+     * @param {float} [options.opacity=1.0] - the b3dm opacity. // unused here for now
+     * @param {boolean|Material=} [options.overrideMaterials='false'] - override b3dm's embedded glTF materials. If
+     * true, a threejs [MeshBasicMaterial](https://threejs.org/docs/index.html?q=meshbasic#api/en/materials/MeshBasicMaterial)
+     * is set up. config.overrideMaterials can also be a threejs [Material](https://threejs.org/docs/index.html?q=material#api/en/materials/Material)
+     * in which case it will be the material used to override.
+     * @return {Promise} - a promise that resolves with an object containig a THREE.Scene (gltf) and a batch table (batchTable).
+     *
+     */
+    parse(buffer, options) {
+        const gltfUpAxisMatrix = options.gltfUpAxisMatrix;
+        const urlBase = options.urlBase;
+        if (!buffer) {
+            throw new Error('No array buffer provided.');
+        }
+
+        const gltfHeader = {};
+
+        // Magic type is unsigned char [4]
+        const magicNumberByteLength = 4;
+        gltfHeader.magic = utf8Decoder.decode(new Uint8Array(buffer, 0, magicNumberByteLength));
+        if (gltfHeader.magic) {
+            const promises = [];
```

Comment:
You don't need an array of promises since there is only one

Reply:
ok